src/report.sh

```diff
@@ -0,0 +1,305 @@
+#
+# @timebox Time box in the format INTEGER[s|m|h]
+#
+# Return:
+# Expanded label in the format INTEGET [seconds|minutes|hours].
+function expand_time_labels()
+{
+  local timebox="$1"
+  local time_type
+  local time_value
+  local time_label
+
+  timebox=$(str_strip "$timebox")
+
+  [[ -z "$timebox" ]] && return 22 # EINVAL
+
+  time_type=$(last_char "$timebox")
+  if [[ ! "$time_type" =~ h|m|s ]]; then
+    time_type='s'
+    timebox="$timebox$time_type"
+  fi
+
+  time_value=$(chop "$timebox")
+  if ! str_is_a_number "$time_value"; then
+    return 22 # EINVAL
+  fi
+
+  case "$time_type" in
+    h)
+      time_label="$time_value hours"
+      ;;
+    m)
+      time_label="$time_value minutes"
+      ;;
+    s)
+      time_label="$time_value seconds"
+      ;;
+  esac
+
+  echo "$time_label"
+}
+
+# Group day data in the tags_details and tags_metadata. Part of the process
+# includes pre-processing raw data in something good to be displayed for users.
+#
+# @day: Day in the format YYYY/MM/DD
+function grouping_day_data()
+{
+  local day="$*"
+  local day_path
+  local details
+  local metadata
+  local start_time
+  local end_time
+  local end_in_sec
+  local timebox
+  local time_label
+
+  day_path=$(join_path "$KW_POMODORO_DATA" "$day")
+
+  # details, total focus time
+  while read -r line; do
+    tag=$(echo "$line" | cut -d ',' -f1)
+    timebox=$(echo "$line" | cut -d ',' -f2)
+    start_time=$(echo "$line" | cut -d ',' -f3)
+    details=$(echo "$line" | cut -d ',' -f4)
+    metadata=$(echo "$line" | cut -d ',' -f 2-)
+
+    time_label=$(expand_time_labels "$timebox")
+    [[ "$?" != 0 ]] && continue
+
+    end_time=$(date --date="$start_time $time_label" +%H:%M:%S)
+
+    [[ -n "$details" ]] && details=": $details"
+    tags_details["$tag"]+=" * [$start_time-$end_time][$timebox]$details\n"
+
+    tags_metadata["$tag"]+="$metadata"
+  done < "$day_path"
+}
+
+# This function groups all week days data.
+#
+# @first_day_of_the_week: First day of the target week
+function grouping_week_data()
+{
+  local first_day_of_the_week="$*"
+  local day_path
+
+  for ((i = 0; i < 7; i++)); do
+    day=$(date --date="${first_day_of_the_week} +${i} day" +%Y/%m/%d)
+    day_path=$(join_path "$KW_POMODORO_DATA" "$day")
+    [[ ! -f "$day_path" ]] && continue
+    grouping_day_data "$day"
+  done
+}
+
+# This function groups all month days data.
+#
+# @target_month: First day of the target month
+function grouping_month_data()
+{
+  local target_month="$*"
+  local month_total_days
+  local day_path
+  local year
+  local month
+  local current_day
+
+  year=$(echo "$target_month" | cut -d '/' -f1)
+  month=$(echo "$target_month" | cut -d '/' -f2)
+  month_total_days=$(days_in_the_month "$month" "$year")
+
+  for ((day = 1; day <= month_total_days; day++)); do
+    current_day="$target_month/$day"
+    day_path=$(join_path "$KW_POMODORO_DATA" "$current_day")
+    [[ ! -f "$day_path" ]] && continue
+    grouping_day_data "$current_day"
+  done
+}
```

Comment:
This ignores all days between 1 and 9, inclusive. `kw pomodoro` creates file names for the days based on the output from `get_today_info`, which returns days with two digits (always, I guess?).

So, for example, if I were to run `kw pomodoro` on July 2nd, it would create a file named "02". By doing this `for` loop above, we are only looking for files named "2" (and of course that applies to the other one-digit numbers too).

Reply:
Super nice catch!